packages/wallets/src/metamask/pages/elements/networkList.element.ts

```diff
@@ -40,4 +59,82 @@ export class NetworkList {
       await this.networkItemBtn.getByText(chainName).click();
     });
   }
+
+  async openModalNetworkEdit(chainId: any) {
+    await this.dialogSection
+      .getByTestId(`network-list-item-options-button-0x${chainId.toString(16)}`)
+      .click();
+    await this.editNetworkButton.click();
+  }
+
+  async isNetworkExist(
+    networkName: string,
+    rpcUrl: string,
+    chainId: number,
+  ): Promise<boolean> {
+    const existNetworkByName = this.dialogSection.getByTestId(networkName);
+    if (await existNetworkByName.isHidden()) {
+      return false;
+    }
+    // that locator exists only for added networks
+    const elements = this.page.getByTestId(
+      `network-rpc-name-button-0x${chainId.toString(16)}`,
+    );
+    const rpcUrlsFound = await elements.filter({ hasText: rpcUrl }).count();
+    return rpcUrlsFound != 0;
+  }
+
+  async addNetworkManually(
+    networkName: string,
+    networkUrl: string,
+    chainId: number,
+    tokenSymbol: string,
+    blockExplorer = '',
+  ) {
+    await test.step('Open the form to add network manually', async () => {
+      await new Header(this.page).networkListButton.click();
```

Comment:
you should to move networkListButton to this class

Reply:
done